xclim/core/indicator.py

```diff
@@ -95,112 +98,144 @@ class Indicator:
+
+    missing = "from_context"
+    missing_options = None
+    context = "none"
+
+    # Variable metadata (_cf_names, those that can be lists or strings)
+    # A developper should access those through _var_attrs on instances
     var_name = None
     standard_name = ""
     long_name = ""
     units = ""
-    missing = "from_context"
-    missing_options = None
     cell_methods = ""
     description = ""
-    context = "none"
+    comment = ""
+
+    # Global metadata (must be strings, not attributed to the output)
     title = ""
     abstract = ""
     keywords = ""
     references = ""
-    comment = ""
     notes = ""
 
     def __new__(cls, **kwds):
-        """Create subclass from arguments."""
-        identifier = kwds.get("identifier", getattr(cls, "identifier"))
+        """Create subclass from arguments. Keywods of kwds that are class attributes are attributed here."""
+        identifier = kwds.get("identifier", cls.identifier)
         if identifier is None:
             raise AttributeError("`identifier` has not been set.")
 
-        # Convert function objects to static methods.
-        for key in cls._funcs + cls._cf_names:
-            if key in kwds and callable(kwds[key]):
-                kwds[key] = staticmethod(kwds[key])
+        kwds["var_name"] = kwds.get("var_name", cls.var_name) or identifier
 
         # Parse `compute` docstring to extract missing attributes
         # Priority: explicit arguments > super class attributes > `compute` docstring info
-
         func = kwds.get("compute", None) or cls.compute
-        attrs = {
-            k: v for (k, v) in parse_doc(func.__doc__).items() if not getattr(cls, k)
-        }
-        attrs.update(kwds)
+        parsed = parse_doc(func.__doc__)
+
+        for name, value in parsed.copy().items():
+            if not getattr(cls, name):
+                # Set if neither the class attr is set nor the kwds attr
+                kwds.setdefault(name, value)
+
+        # Convert function objects to static methods.
+        for key in cls._funcs:
+            if key in kwds and callable(kwds[key]):
+                kwds[key] = staticmethod(kwds[key])
+
+        # Get number of outputs
+        n_outs = (
+            len(kwds["var_name"]) if isinstance(kwds["var_name"], (list, tuple)) else 1
+        )
+        # Populate _var_attrs from attribute set during class creation and __new__
+        kwds["cf_attrs"] = [{} for i in range(n_outs)]
+        for name in cls._cf_names:
+            values = kwds.get(name, getattr(cls, name))
+            if not isinstance(values, (list, tuple)):
+                values = [values] * n_outs
+            elif len(values) != n_outs:
+                raise ValueError(
+                    f"Attribute {name} has {len(values)} elements but should have {n_outs} according to passed var_name."
+                )
+            for attrs, value in zip(kwds["cf_attrs"], values):
+                if value:
+                    attrs[name] = value
```

Comment:
Does it make sense to put this logic into a method ?

Reply:
Could. And it would reduce the complexity of the method...